main_2024.py

```diff
@@ -0,0 +1,166 @@
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--cpu", action="store_true", help="option to force cpu")
+    args = parser.parse_args()
+
+    # Set constants
+    try:
+        QUEUE_MAX_SIZE = config["queue_max_size"]
+
+        VIDEO_INPUT_CAMERA_NAME = config["video_input"]["camera_name"]
+        VIDEO_INPUT_WORKER_PERIOD = config["video_input"]["worker_period"]
+        VIDEO_INPUT_SAVE_PREFIX = config["video_input"]["save_prefix"]
+
+        DETECT_TARGET_WORKER_COUNT = config["detect_target"]["worker_count"]
+        DETECT_TARGET_DEVICE =  "cpu" if args.cpu else config["detect_target"]["device"]
+        DETECT_TARGET_MODEL_PATH = config["detect_target"]["model_path"]
+        DETECT_TARGET_SAVE_PREFIX = config["detect_target"]["save_prefix"]
+
+        FLIGHT_INTERFACE_ADDRESS = config["flight_interface"]["address"]
+        FLIGHT_INTERFACE_WORKER_PERIOD = config["flight_interface"]["worker_period"]
+    except KeyError:
+        print("Config key(s) not found")
+        return -1
+
+    # Setup
+    controller = worker_controller.WorkerController()
+
+    mp_manager = mp.Manager()
+    video_input_to_detect_target_queue = queue_proxy_wrapper.QueueProxyWrapper(
+        mp_manager,
+        QUEUE_MAX_SIZE,
+    )
+    detect_target_to_main_queue = queue_proxy_wrapper.QueueProxyWrapper(
+        mp_manager,
+        QUEUE_MAX_SIZE,
+    )
+    flight_interface_to_data_merge_queue = queue_proxy_wrapper.QueueProxyWrapper(
+        mp_manager,
+        QUEUE_MAX_SIZE
+    )
+
+    video_input_manager = worker_manager.WorkerManager()
+    video_input_manager.create_workers(
+        1,
+        video_input_worker.video_input_worker,
+        (
+            VIDEO_INPUT_CAMERA_NAME,
+            VIDEO_INPUT_WORKER_PERIOD,
+            VIDEO_INPUT_SAVE_PREFIX,
+            video_input_to_detect_target_queue,
+            controller,
+        ),
+    )
+
+    detect_target_manager = worker_manager.WorkerManager()
+    detect_target_manager.create_workers(
+        DETECT_TARGET_WORKER_COUNT,
+        detect_target_worker.detect_target_worker,
+        (
+            DETECT_TARGET_DEVICE,
+            DETECT_TARGET_MODEL_PATH,
+            DETECT_TARGET_SAVE_PREFIX,
+            video_input_to_detect_target_queue,
+            detect_target_to_main_queue,
+            controller,
+        ),
+    )
+
+    flight_interface_manager = worker_manager.WorkerManager()
+    flight_interface_manager.create_workers(
+        1,
+        flight_interface_worker.flight_input_worker,
+        (
+            FLIGHT_INTERFACE_ADDRESS,
+            FLIGHT_INTERFACE_WORKER_PERIOD,
+            flight_interface_to_data_merge_queue,
+            controller
+        ),
+    )
+
+    # Run
+    video_input_manager.start_workers()
+    detect_target_manager.start_workers()
+    flight_interface_manager.start_workers()
+
+    while True:
+        image = detect_target_to_main_queue.queue.get()
```

Comment:
Since the flight interface worker rate is much faster than the detect target rate (which is limited by the video input rate), you can use `get_nowait()` with the detect target queue, and enclose it with a try catch block.